### ISSUE TITLE
Error running CyLR.exe

### ISSUE THREAD

**Issue by barrie0482** [orlikoski/CyLR#78]
Hi,
I have just started trying this tool out and have run into a problem. I noticed in the zip file produced by the CyLR.exe The Google Chrome history file is a directory. Not the History file I expected.

I am after the whole Google Chrome Default directory, files and sub-directories, so I tried the following.

D:\tmp\fred>CyLR_win-x64\CyLR.exe -c "C:/Users/fred/AppData/Local/Google/Chrome/User Data/Default"
Error: Could not find file: C:/Users/fred/AppData/Local/Google/Chrome/User Data/Default
Exiting
Error occured while collecting files:
System.ArgumentException: Value does not fall within the expected range.
   at CyLR.CollectionPaths.GetPaths(Arguments arguments, List`1 additionalPaths) in /home/travis/build/orlikoski/CyLR/CyLR/src/CollectionPaths.cs:line 167
   at CyLR.Program.Main(String[] args) in /home/travis/build/orlikoski/CyLR/CyLR/src/Program.cs:line 57
   
 I am using CyLR Version 2.0.0.0, running as administrator, windows 10 Home ver 1809. The path to the 
 Google Chrome Default directory on my computer is correct.
 
 Any suggestions on what I have done wrong?
 
 Thanks

Cheers, Barrie

**Comment by orlikoski** [comment]
The -c flag is for sending a file of file paths, the space delimited file paths as arguments do not require -c.

Try `D:\tmp\fred>CyLR_win-x64\CyLR.exe "C:/Users/fred/AppData/Local/Google/Chrome/User Data/Default"`

**Comment by barrie0482** [comment]
Good Day Alan,

Happy New Year to you.

Thank you for replying to my question.

I tried the command without the "-c" option as you suggested and I was able to get the contents of the Google Chrome Default folder successfully.
I also tried the command with the "-c" option and a config file and I was able to get the contents of the Google Chrome Default folder successfully. I suspect that I will be using a custom config file as there are a couple of other artefacts that I sometimes gather.

When I am collecting data for a triage image, I normally just take the whole of "C:\Users\<username>\AppData\Local\Google\Chrome\User Data\Default\" and use Hindsight to parse the data in the collected "Default" directory.

In the source file "CollectionPaths.cs" in line 90, the History file is followed by a backslash, which appears to make it a directory instead of a file. In the zip archive that is created on my computer, an empty folder is created instead of the History file. Should line 90 in CollectionPaths.cs contain 

defaultPaths.Add($@"{user.ProfilePath}\AppData\Local\Google\Chrome\User Data\Default\"

instead of

defaultPaths.Add($@"{user.ProfilePath}\AppData\Local\Google\Chrome\User Data\Default\History\"

Thanks.

Cheers, Barrie

**Comment by barrie0482** [comment]
Good Day Alan,

I have modified line 90 in source file "CollectionPaths.cs" to reflect the following line.

defaultPaths.Add($@"{user.ProfilePath}\AppData\Local\Google\Chrome\User Data\Default"

I compiled the modified source and ran the new binary. All of the contents of the Google Chrome "Default" directory are now included in the output zip file.
I'll continue testing CyLR.  I have been looking for a tool like this for a while.

Thanks.

Cheers, Barrie
